src/cuda/1d/spread1d_wrapper.cu

```diff
@@ -50,10 +51,30 @@ int cuspread1d(cufinufft_plan_t<T> *d_plan, int blksize)
   return ier;
 }
 
+template<typename T> struct cmp : public thrust::binary_function<int, int, bool> {
+
+  cmp(const T *kx) : kx(kx) {}
+
+  __host__ __device__ bool operator()(const int a, const int b) const {
+    return fold_rescale(kx[a], 1) < fold_rescale(kx[b], 1);
+  }
+
+private:
+  const T *kx;
+};
+
 template<typename T>
 int cuspread1d_nuptsdriven_prop(int nf1, int M, cufinufft_plan_t<T> *d_plan) {
   auto &stream = d_plan->stream;
-
+  if (d_plan->opts.gpu_sort && d_plan->opts.gpu_method == 1) {
+    int *d_idxnupts = d_plan->idxnupts;
+    thrust::sequence(thrust::cuda::par.on(stream), d_idxnupts, d_idxnupts + M);
+    RETURN_IF_CUDA_ERROR
+    thrust::sort(thrust::cuda::par.on(stream), d_idxnupts, d_idxnupts + M,
```

Comment:
Does thrust sort will also be faster than current bin sort in 2D and 3D? Though sort only takes few percentage in 2D and 3D.

One thing to note is that thrust sort(most probably calls cub sort) will create a workspace during sorting, so the GPU memory may have a little spike, while current binsort's memory is all managed by ourselves.